streams-tests/jvm/src/test/scala/zio/stream/SinkSpec.scala

```diff
@@ -167,6 +167,9 @@ class SinkSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
     zipLeft (<*)
       happy path $zipLeftHappyPath
 
+    zipPar
+       zipPar somehow works $zipParSanity
```

Comment:
Here are some additional test cases I suggest we add:
- left sink finishes earlier
- right sink finishes earlier
- both sinks finish at the same time
- left sink errors during step
- left sink errors during extract
- right sink errors during step
- right sink errors during extract

Reply:
Yeah. Must cover cases of non-terminating sinks as well. And also it would be nice to explicitly check the logic of choosing the shorter `leftover` when sinks terminated at the same time.
Haven't found test sinks with non-trivial leftovers to use here. Will invent one.